src/sys.d.ts

```diff
@@ -0,0 +1,1352 @@
+  args(): List<string>
+  mainMethod(): Method | null
+  vars(): Map<string, string>
+  diagnostics(): Map<string, JsObj>
+  gc(): void
+  host(): string
+  user(): string
+  in$(): InStream
+  out(): OutStream
+  err(): OutStream
+  prompt(msg?: string): string | null
+  promptPassword(msg?: string): string | null
+  homeDir(): File
+  workDir(): File
+  tempDir(): File
+  path(): List<File>
+  findFile(uri: Uri, checked?: boolean): File | null
+  findAllFiles(uri: Uri): List<File>
+  findPodFile(podName: string): File | null
+  findAllPodNames(): List<string>
+  compileScript(f: File, options?: Map<string, JsObj> | null): Type
+  compileScriptToJs(f: File, options?: Map<string, JsObj> | null): string
+  index(key: string): List<string>
+  indexKeys(): List<string>
+  indexPodNames(key: string): List<string>
+  props(pod: Pod, uri: Uri, maxAge: Duration): Map<string, string>
+  config(pod: Pod, key: string, def?: string | null): string | null
+  locale(pod: Pod, key: string, def?: string | null, locale?: Locale): string | null
+  exit(status?: number): void
+  addShutdownHook(hook: (() => void)): void
+  removeShutdownHook(hook: (() => void)): boolean
+}
+export class Map<K = unknown, V = unknown> extends Obj {
+  caseInsensitive(): boolean
+  caseInsensitive(it: boolean): void
+  ordered(): boolean
+  ordered(it: boolean): void
+  def(): V | null
+  def(it: V | null): void
+  static make(type: Type): Map
+  isEmpty(): boolean
+  size(): number
+  get(key: K, def?: V | null): V | null
+  getChecked(key: K, checked?: boolean): V | null
+  getOrThrow(key: K): V
+  containsKey(key: K): boolean
+  keys(): List<K>
+  vals(): List<V>
+  dup(): Map<K,V>
+  set(key: K, val: V): Map<K,V>
+  add(key: K, val: V): Map<K,V>
+  addIfNotNull(key: K, val: V | null): Map<K,V>
+  addNotNull(key: K, val: V | null): Map<K,V>
+  getOrAdd(key: K, valFunc: ((argA: K) => V)): V
+  setAll(m: Map<K,V>): Map<K,V>
+  addAll(m: Map<K,V>): Map<K,V>
+  setList(list: List<V>, c?: ((argA: V, argB: number) => K) | null): Map<K,V>
+  addList(list: List<V>, c?: ((argA: V, argB: number) => K) | null): Map<K,V>
+  remove(key: K): V | null
+  clear(): Map<K,V>
+  join(separator: string, c?: ((argA: V, argB: K) => string) | null): string
+  toCode(): string
+  each(c: ((argA: V, argB: K) => void)): void
+  eachWhile(c: ((argA: V, argB: K) => JsObj | null)): JsObj | null
+  find(c: ((argA: V, argB: K) => boolean)): V | null
+  findAll(c: ((argA: V, argB: K) => boolean)): Map<K,V>
+  findNotNull(): Map<K,V>
+  exclude(c: ((argA: V, argB: K) => boolean)): Map<K,V>
+  any(c: ((argA: V, argB: K) => boolean)): boolean
+  all(c: ((argA: V, argB: K) => boolean)): boolean
+  reduce(init: JsObj | null, c: ((argA: JsObj | null, argB: V, argC: K) => JsObj | null)): JsObj | null
+  map(c: ((argA: V, argB: K) => JsObj | null)): Map<JsObj, JsObj | null>
+  mapNotNull(c: ((argA: V, argB: K) => JsObj | null)): Map<JsObj, JsObj | null>
+  isRO(): boolean
+  isRW(): boolean
+  ro(): Map<K,V>
```

Comment:
For collections that return readonly types, the canonical way in ts would be: `Readonly<Map<K,V>>`